### PR TITLE
Feed Optimizations

### PR DESCRIPTION
**Pull request by le-sun**
Using django-filters for filtering document feed rather than our own custom logic
- This should improve readability and general usage

Optimized filtering queries
- Around a 15-20% reduction in SQL queries

Code base cleanup

### CODE REVIEW

File: src/researchhub_document/views/custom/unified_document_pagination.py [hunk 1]
```diff
@@ -1,10 +1,33 @@
...
+            "SELECT reltuples FROM pg_class WHERE relname = 'researchhub_document_researchhubunifieddocument'"
+        )
+        n = int(cursor.fetchone()[0])
+        return n
+
+    def get_paginated_response(self, data):
+        return Response(
+            OrderedDict(
+                [
+                    ("count", self._get_estimated_count()),
```
Comment (le-sun) [inline src/researchhub_document/views/custom/unified_document_pagination.py:27]: This is more of a temporary solution to our pagination issue. Is retrieves an estimated count of items, which is much faster than a sequence scan for a full count.

File: src/researchhub_document/filters.py [hunk 2]
```diff
@@ -0,0 +1,250 @@
+from django.db.models import Count, Q
```
Comment (yattias) [inline src/researchhub_document/filters.py:1]: Nice moving it to its own file 💯

File: src/researchhub_document/serializers/researchhub_unified_document_serializer.py [hunk 3]
```diff
@@ -97,7 +97,7 @@ class Meta:
         fields = "__all__"
 
     def get_featured(self, unified_doc):
-        hub_id = self.context.get("hub_id")
+        hub_id = self.context.get("hub_id", 0)
```
Comment (yattias) [inline src/researchhub_document/serializers/researchhub_unified_document_serializer.py:100]: Nice

File: src/researchhub_document/views/researchhub_unified_document_views.py [hunk 4]
```diff
@@ -44,28 +36,42 @@
...
+        return Response(status=403)
+
+    def retrieve(self, *args, **kwargs):
+        return Response(status=403)
+
+    def partial_update(self, *args, **kwargs):
+        return Response(status=403)
+
+    def destroy(self, *args, **kwargs):
+        return Response(status=403)
```
[the remark this answers is not in the document]
Comment (yattias) [reply src/researchhub_document/views/researchhub_unified_document_views.py:73]: Nice

**Review by yattias — APPROVED** [review]
LGTM just needs a bunch of testing

**Pull request merged**